sql-plugin/src/main/spark332cdh/scala/com/nvidia/spark/rapids/shims/SparkShims.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "332cdh"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids.shims
+
+import com.nvidia.spark.rapids._
+
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.command.{CreateDataSourceTableAsSelectCommand, DataWritingCommand, RunnableCommand}
+
+object SparkShimImpl extends Spark331PlusShims with AnsiCastRuleShims {
+
+  override def getDataWriteCmds: Map[Class[_ <: DataWritingCommand],
+      DataWritingCommandRule[_ <: DataWritingCommand]] = {
+    Seq(GpuOverrides.dataWriteCmd[CreateDataSourceTableAsSelectCommand](
+    "Create table with select command",
+    (a, conf, p, r) => new CreateDataSourceTableAsSelectCommandMeta(a, conf, p, r))
+    ).map(r => (r.getClassFor.asSubclass(classOf[DataWritingCommand]), r)).toMap
+  }
+
+  override def getRunnableCmds: Map[Class[_ <: RunnableCommand],
+      RunnableCommandRule[_ <: RunnableCommand]] = {
+    Map.empty
+  }
+
+  override def getExecs: Map[Class[_ <: SparkPlan], ExecRule[_ <: SparkPlan]] = {
+
+    val newRule = org.apache.spark.sql.hive.cdh.HiveOverrides.getHiveTableExecRule()
+    // scala compiler gets confused if we don't give an explicit type to the class here.
+    // ... perhaps there is a cleaner way to do this
+    val ruleClass: Class[_ <: SparkPlan] = newRule.getClassFor.asSubclass(classOf[SparkPlan])
+    super.getExecs ++ Seq(ruleClass -> newRule).toMap
+  }
```

Comment:
The three methods in the body of SparkShimImpl for 330cdh and 332cdh are identical.

We could have another trait 
```Scala
/*** spark-rapids-shim-json-lines
{"spark": "330cdh"}
{"spark": "332cdh"}
spark-rapids-shim-json-lines ***/
package com.nvidia.spark.rapids.shims

trait Spark33cdhShims extends AnsiCastRuleShims {
  override def getDataWriteCmds: Map[Class[_ <: DataWritingCommand],
      DataWritingCommandRule[_ <: DataWritingCommand]] = {
    Seq(GpuOverrides.dataWriteCmd[CreateDataSourceTableAsSelectCommand](
    "Create table with select command",
    (a, conf, p, r) => new CreateDataSourceTableAsSelectCommandMeta(a, conf, p, r))
    ).map(r => (r.getClassFor.asSubclass(classOf[DataWritingCommand]), r)).toMap
  }

  override def getRunnableCmds: Map[Class[_ <: RunnableCommand],
      RunnableCommandRule[_ <: RunnableCommand]] = {
    Map.empty
  }

  override def getExecs: Map[Class[_ <: SparkPlan], ExecRule[_ <: SparkPlan]] = {

    val newRule = org.apache.spark.sql.hive.cdh.HiveOverrides.getHiveTableExecRule()
    // scala compiler gets confused if we don't give an explicit type to the class here.
    // ... perhaps there is a cleaner way to do this
    val ruleClass: Class[_ <: SparkPlan] = newRule.getClassFor.asSubclass(classOf[SparkPlan])
    super.getExecs ++ Seq(ruleClass -> newRule).toMap
  }
}
```

for 330cdh `object SparkShimImpl extends Spark330PlusNonDBShims with Spark33cdhShims`

for 332cdh `object SparkShimImpl extends Spark331PlusShims with Spark33cdhShims`

Reply:
updated but slightly differently to deal with calling super.getExecs from trait